### PR TITLE
Update for Swift 3.1

### PR DESCRIPTION
**Pull request by farice**
Mainly syntactical changes.

### CODE REVIEW

File: Classes/CocoaLumberjack.swift [hunk 1]
```diff
@@ -52,39 +54,44 @@ public func resetDefaultDebugLevel() {
...
-    _DDLogMessage(message, level: level, flag: .verbose, context: context, file: file, function: function, line: line, tag: tag, asynchronous: async, ddlog: ddlog)
+public func DDLogVerbose( message: @autoclosure () -> String, level: DDLogLevel = defaultDebugLevel, context: Int = 0, file: StaticString = #file, function: StaticString = #function, line: UInt = #line, tag: AnyObject? = nil, asynchronous async: Bool = true, ddlog: DDLog = DDLog.sharedInstance()) {
+    _DDLogMessage(message: message, level: level, flag: .verbose, context: context, file: file, function: function, line: line, tag: tag, asynchronous: async, ddlog: ddlog)
 }
 
-public func DDLogError(_ message: @autoclosure () -> String, level: DDLogLevel = defaultDebugLevel, context: Int = 0, file: StaticString = #file, function: StaticString = #function, line: UInt = #line, tag: Any? = nil, asynchronous async: Bool = false, ddlog: DDLog = DDLog.sharedInstance) {
```
Comment (rivera-ernesto) [inline Classes/CocoaLumberjack.swift]: Why remove unnamed parameters?

**Comment by sushichop** [comment]
IMO, CocoaLumberjack 3.2.0 already supports Swift 3.1.

And why do we need to remove unnamed parameters as @rivera-ernesto  says ?
I think the unnamed parameter at 1st argument is for a common swifty style, so I think we should not remove it...

**Comment by farice** [comment]
You are right. Unnamed parameters as a 1st argument is mainstream Swifty style. I reverted those changes.

File: Classes/CocoaLumberjack.swift [hunk 2]
```diff
@@ -16,28 +16,30 @@
 import Foundation
 
 extension DDLogFlag {
-    public static func from(_ logLevel: DDLogLevel) -> DDLogFlag {
+    public static func fromLogLevel(_ logLevel: DDLogLevel) -> DDLogFlag {
```
Comment (rivera-ernesto) [inline Classes/CocoaLumberjack.swift]: Agree that here `from(logLevel: DDLogLevel)` would be more "Swift".

File: Classes/CocoaLumberjack.swift [hunk 3]
```diff
@@ -52,39 +54,44 @@ public func resetDefaultDebugLevel() {
...
+        ddlog.log(asynchronous, message: logMessage)
     }
 }
 
-public func DDLogDebug(_ message: @autoclosure () -> String, level: DDLogLevel = defaultDebugLevel, context: Int = 0, file: StaticString = #file, function: StaticString = #function, line: UInt = #line, tag: Any? = nil, asynchronous async: Bool = true, ddlog: DDLog = DDLog.sharedInstance) {
-    _DDLogMessage(message, level: level, flag: .debug, context: context, file: file, function: function, line: line, tag: tag, asynchronous: async, ddlog: ddlog)
+public func DDLogDebug(_ message: @autoclosure () -> String, level: DDLogLevel = defaultDebugLevel, context: Int = 0, file: StaticString = #file, function: StaticString = #function, line: UInt = #line, tag: AnyObject? = nil, asynchronous async: Bool = true, ddlog: DDLog = DDLog.sharedInstance()) {
+    _DDLogMessage(message: message, level: level, flag: .debug, context: context, file: file, function: function, line: line, tag: tag, asynchronous: async, ddlog: ddlog)
```
Comment (rivera-ernesto) [inline Classes/CocoaLumberjack.swift]: I think now `Any` is preferred over `AnyObject`.

**Comment by rivera-ernesto** [comment]
3.3.0 now supports Swift 4.

**Pull request closed without merging**